UnityProject/Assets/Scripts/Shuttles/NetworkedMatrixMove.cs

```diff
@@ -0,0 +1,1438 @@
+		{
+			foreach (var Thruster in ConnectedThrusters)
+			{
+				if (Thruster.ThisThrusterDirectionClassification != Thruster.ThrusterDirectionClassification.Up &&
+				    Thruster.ThisThrusterDirectionClassification != Thruster.ThrusterDirectionClassification.Down)
+				{
+					Thruster.TargetMolesUsed = Thruster.MaxMolesUseda * 0;
+				}
+			}
+
+
+			if (Multiplier < 0.9f) return;
+			var CurrentOrientation = TargetTransform.eulerAngles.z.Angle360ToOrientationEnum();
+
+			if (Direction == Thruster.ThrusterDirectionClassification.Right)
+			{
+				switch (CurrentOrientation)
+				{
+					case OrientationEnum.Up_By0:
+						TargetOrientation = OrientationEnum.Right_By270;
+						break;
+					case OrientationEnum.Right_By270:
+						TargetOrientation = OrientationEnum.Down_By180;
+						break;
+					case OrientationEnum.Down_By180:
+						TargetOrientation = OrientationEnum.Left_By90;
+						break;
+					case OrientationEnum.Left_By90:
+						TargetOrientation = OrientationEnum.Up_By0;
+						break;
+				}
+			}
+			else
+			{
+				switch (CurrentOrientation)
+				{
+					case OrientationEnum.Up_By0:
+						TargetOrientation = OrientationEnum.Left_By90;
+						break;
+					case OrientationEnum.Left_By90:
+						TargetOrientation = OrientationEnum.Down_By180;
+						break;
+					case OrientationEnum.Down_By180:
+						TargetOrientation = OrientationEnum.Right_By270;
+						break;
+					case OrientationEnum.Right_By270:
+						TargetOrientation = OrientationEnum.Up_By0;
+						break;
+				}
+			}
+		}
+	}
+
+	public void AddConnector(ShuttleConnector ShuttleConnector)
+	{
+		if (ConnectedShuttleConnectors.Contains(ShuttleConnector) == false)
+		{
+			ConnectedShuttleConnectors.Add(ShuttleConnector);
+		}
+	}
+
+
+	public void RemoveConnector(ShuttleConnector ShuttleConnector)
+	{
+		if (ConnectedShuttleConnectors.Contains(ShuttleConnector))
+		{
+			ConnectedShuttleConnectors.Remove(ShuttleConnector);
+		}
+	}
+
+
+	public void AddThruster(Thruster Thruster)
+	{
+		if (ConnectedThrusters.Contains(Thruster) == false)
+		{
+			ConnectedThrusters.Add(Thruster);
+		}
+	}
+
+
+	public void RemoveThruster(Thruster Thruster)
+	{
+		if (ConnectedThrusters.Contains(Thruster))
+		{
+			ConnectedThrusters.Remove(Thruster);
+		}
+	}
+
+	public void RcsMove(Orientation GlobalMoveDirection)
+	{
+		RcsMove(GlobalMoveDirection.LocalVector.ToOrientationEnum());
+	}
+
+	public void RcsMove(OrientationEnum GlobalMoveDirection)
+	{
+		if (IsMoving) return;
+
+		bool HasThrusterDirection = false;
+
+		if (RCSRequiresThrusters)
+		{
+			foreach (var Thruster in ConnectedThrusters)
+			{
+				if (((Vector3) Thruster.Rotatable.WorldDirection).ToOrientationEnum() ==
+				    GlobalMoveDirection.ToOpposite())
+				{
+					HasThrusterDirection = true;
+					break;
+				}
+			}
+		}
+		else
+		{
+			HasThrusterDirection = true;
+		}
+
+
+		if (HasThrusterDirection)
+		{
+			WorldCurrentVelocity += GlobalMoveDirection.ToLocalVector3();
+		}
+	}
+
+
+	public void UpdateLoop()
```

Comment:
for the love of god man stop creating monolithic functions.
Reduce the complexity of this